search-provider/eks-discovery-feed-provider.c

```diff
@@ -435,6 +469,166 @@ handle_article_card_descriptions (EksDiscoveryFeedDatabaseContentProvider *skele
 }
 
 static void
+get_word_of_the_day_content_cb (GObject *source,
+                                GAsyncResult *result,
+                                gpointer user_data)
+{
+  EkncEngine *engine = EKNC_ENGINE (source);
+  DiscoveryFeedQueryState *state = user_data;
+
+  g_application_release (g_application_get_default ());
+
+  GError *error = NULL;
+  g_autoptr(GSList) models = NULL;
+  GSList *shards = NULL;
+
+  if (!models_and_shards_for_result (engine,
+                                     state->provider->application_id,
+                                     result,
+                                     &models,
+                                     &shards,
+                                     &error))
+    {
+      g_dbus_method_invocation_take_error (state->invocation, error);
+      discovery_feed_query_state_free (state);
+      return;
+    }
+
+  GVariantBuilder builder;
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{ss}"));
+
+  gint index = get_day_of_year () % g_slist_length (models);
+  EkncContentObjectModel *model = g_slist_nth (models, index)->data;
+
+  add_key_value_pair_from_model_to_variant (model, &builder, "title");
+  add_key_value_pair_from_model_to_variant (model, &builder, "synopsis");
+  add_key_value_pair_from_model_to_variant (model, &builder, "ekn-id");
```

Comment:
Note that you will also need `pronunciation` `definition` and `word_type`. Not sure if the ingesters are producing those fields ATM, or if we need to stuff them into `discoveryFeedContent` and read them out here.